packages/flutter/test/material/tabs_test.dart

```diff
@@ -4378,6 +4378,22 @@ void main() {
     expect(() => Tab(text: 'foo', child: Container()), throwsAssertionError);
   });
 
+  test('Tab throws clear error when both text and child are set', () {
+    // Wrap in a closure so the assertion is checked at runtime
+    expect(
+      () {
+        Tab(text: 'Hi', child: const Text('World')); // no const
+      },
+      throwsA(
+        const TypeMatcher<AssertionError>().having(
+          (AssertionError error) => error.message,
+          'message',
+          contains('Provide either text or child, not both, when creating a Tab.'),
+        ),
+      ),
+    );
+  });
+
```

Comment:
Nit: Test the other error message as well? Or is that already covered in an existing test?